listings/ch01-advanced-concepts/storing_arrays/src/storing_arrays.cairo

```diff
@@ -73,26 +69,22 @@ impl StorageAccessFelt252Array of StorageAccess<Array<felt252>> {
         }
     }
 
-    fn size_internal(value: Array<felt252>) -> u8 {
-        if value.len() == 0 {
-            return 1;
-        }
-        1_u8 + StorageAccess::<felt252>::size_internal(*value[0]) * value.len().try_into().unwrap()
+    fn size() -> u8 {
```

Comment:
@enitrat Do you know how to upgrade this part to 2.1.0?
The `size` function don't take a `value` arg anymore and the size is fixed by type.

Reply:
yeah this is a bit troublesome, I can't think of any workaround for this. Perhaps `size` could return the maximum value, so `255`?

Reply:
I agree that returning the upper bound is a solution. This is however a bit confusing with an `array` type.

Reply:
this question has been asked quite a lot and I don't think we received an answer from the compiler team